### PR TITLE
Implement BaseAPIClient in http-client package

### PR DESCRIPTION
**Pull request by RonaldEAM**

### CODE REVIEW

File: packages/integration-sdk-http-client/src/errors.ts [hunk 1]
```diff
@@ -1,17 +1,124 @@
...
+      endpoint,
+      retryAfter: Number(response.headers.get('retry-after')),
+    });
+  }
+
+  return new RetryableIntegrationProviderApiError({
+    cause: new HTTPResponseError(response),
+    endpoint,
+    status: response.status,
+    statusText: response.statusText ?? response.status,
```
Comment (i5o) [inline packages/integration-sdk-http-client/src/errors.ts:76]:
```suggestion
    statusText: response.statusText ?? undefined,
```

```suggestion
    statusText: response.statusText,
```

**Comment by i5o** [comment]
Looks good to me so far.
What happens if a request returns 302/301, do we follow the URL?

File: packages/integration-sdk-http-client/package.json [hunk 2]
```diff
@@ -23,12 +23,14 @@
     "prepack": "yarn build:dist"
   },
   "dependencies": {
+    "@lifeomic/attempt": "^3.0.3",
+    "lodash": "^4.17.21",
```
[the remark this answers is not in the document]
Comment (RonaldEAM) [reply packages/integration-sdk-http-client/package.json:27]: lodash discourage the use of those packages (https://lodash.com/per-method-packages), I'm importing the method only `import get from 'lodash/get';`, according to them "only the subset of lodash code your package uses will be bundled in projects that use your package."

File: packages/integration-sdk-http-client/src/client.ts [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+};
+
+const DEFAULT_RATE_LIMIT_HEADERS: { [key in keyof RateLimitHeaders]: string } =
+  {
+    limit: 'x-rate-limit-limit',
+    remaining: 'x-rate-limit-remaining',
+    reset: 'x-rate-limit-reset',
+  };
+
+export abstract class BaseAPIClient {
```
[the remark this answers is not in the document]
Comment (RonaldEAM) [reply packages/integration-sdk-http-client/src/client.ts:64]: I added an optional token bucket similar following the implementation in template-engine, I think project bocchi should use this base client so it should support everything in the template

**Review by i5o — APPROVED** [review]

**Review by JakeFerrero — APPROVED** [review]

**Pull request merged**